### PR TITLE
feat(directive): add ng-switch directive

### PR DESCRIPTION
**Pull request by vicb**
TODO:
- [x] finish the impl (default + todos)
- [x] add tests
- [x] add inline doc

### CODE REVIEW

**Comment by vicb** [comment]
@mhevery for now i have injected `NgElement` to read the attribute value because `ng-switch-when` value should be a string (it is not an expression).

If I add `bind: { 'ng-switch-when': 'when'}` then the attribute value will be interpreted as an expression and watched. I don't think this is something we want.

**Comment by mhevery** [comment]
@vicb after talking about it with the team, we have just decided that the way template works is fine. This implies that `foo` has to be quoted `'foo'` in order to be a string constant. The added benefit would can switch on numbers as well as booleans. The implication is that yes the case can change its value due to data binding, but we think it is fine.

**Comment by vicb** [comment]
@mhevery I'll update the PR according to your latest comment.

File: modules/directives/test/ng_switch_spec.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    function compileWithTemplate(template) {
+      return compiler.compile(TestComponent, createElement(template));
+    }
+
+    describe('switch value changes', () => {
+      it('should switch amongst when values', (done) => {
+        var template = '<div>' +
+          '<ul [ng-switch]="switchValue">' +
+            '<template [ng-switch-when]="\'a\'"><li>when a</li></template>' +
+            '<template [ng-switch-when]="\'b\'"><li>when b</li></template>' +
```
Comment (benouat) [inline modules/directives/test/ng_switch_spec.js]: I think all your templates inside this test file are missing the closing `</ul>`

Comment (vicb) [inline modules/directives/test/ng_switch_spec.js]: good catch, thank you. I'll update.

Comment (pkozlowski-opensource) [inline modules/directives/test/ng_switch_spec.js]: @vicb don't we have dangling `<div>`s now?
https://github.com/vicb/angular/commit/92c580e5d3b7de34685b058c2ac6eb7ae45e3b90#diff-e974709717f129a9569c4445c2ba0f54R37

Comment (vicb) [inline modules/directives/test/ng_switch_spec.js]: thanks!
I hope I finally got it right!

**Comment by vicb** [comment]
Does removed view need to be dehydrated ? check #416

**Pull request merged**